hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -544,15 +543,40 @@ public static Option<Pair<Integer, String>> getLatestLogVersion(FileSystem fs, P
   }
 
   /**
-   * computes the next log version for the specified fileId in the partition path.
+   * Get the latest log version for the fileId in the partition path.
+   */
+  public static int getLatestLogVersion(FileSystem fs, Path partitionPath, final String fileId,
+                                                        final String logFileExtension, final String deltaCommitTime, final String writeToken) throws IOException {
+    int version = HoodieLogFile.LOGFILE_BASE_VERSION;
+    String logFileName = makeLogFileName(fileId, logFileExtension, deltaCommitTime, version, writeToken);
+    while (fileExists(fs, new Path(partitionPath, logFileName))) {
```

Comment:
would like to avoid these fs calls. Can't we have the log file versions be used locally generated by writers, since the delta commit time is anyway in the log file name to differentiate?

Reply:
You are right, actually we already do that, in the `LogFormatWriter` we would check the file existence anyway when creating a new file, and here we also do a check for generating the next version number, in `LogFormatWriter`, can maintain the latest used version number there, so each time we start from the latest number.

You mean to remove the file existence check totally, I'm fine with that.

Reply:
Yes, we should not have the need to do exists check in `HoodieLogFormatWriter.getOutputStream`. Go ahead with creating new log file, it it already exists (which will be known the file system exception thrown upon creation), just rollover. Let's remove all `fileExists` usages.

Reply:
Yeah, +1